### ISSUE TITLE
Harvest Hours Today and Harvest Hours Week are always 0

### ISSUE THREAD

**Issue by hco** [BottlecapDave/HomeAssistant-HarvestTimeTracker#10]
### Describe the bug

I managed to setup the integration, the default task select has options, so the communication seems to work, but the total hours for the week and for today are always 0. The also don't contain entries in the attributes.
![CleanShot 2024-08-01 at 12 59 43@2x](https://github.com/user-attachments/assets/a4a942c3-6700-47db-9407-80810fe5fd53)

### Reproduction steps

-

### Expected behaviour

I expected the state to be bigger than 0, as I have been working for a few hours today already

### Integration Version

1.1.2

### Home Assistant Version

2024.7.4

### Fresh Install?

Fresh install

### Home Assistant Logs

```
2024-08-01 13:09:40.414 DEBUG (MainThread) [custom_components.harvest_time_tracker.coordinators.time_entries] Failed to retrieve time entries
2024-08-01 13:09:40.415 DEBUG (MainThread) [custom_components.harvest_time_tracker.coordinators.time_entries] Finished fetching time_entries data in 0.608 seconds (success: True)
2024-08-01 13:09:58.858 DEBUG (MainThread) [custom_components.harvest_time_tracker.coordinators.time_entries] Failed to retrieve time entries
2024-08-01 13:09:58.859 DEBUG (MainThread) [custom_components.harvest_time_tracker.coordinators.time_entries] Finished fetching time_entries data in 0.584 seconds (success: True)
```

**Comment by BottlecapDave** [comment]
Hello and sorry you're seeing this issue. The errors state it's failing to retrieve the hours, but not giving a reason. I'll see if I can add more logging over the weekend to try and track down the reason. Out of interest, do you know what hours you should be seeing so I can try and recreate the issue.

**Comment by hco** [comment]
Yesterday I should have seen ~5:15, at the time of writing the report it should have been ~4 hours.

You can also point me to the lines of code where that make the request to the API, I could print the query parameters and check if I get entries when I make the request manually.

--
Edit: Now that I think about it, adding debug logs that contain the request URL if fetching entries failed could be useful for debugging problems of others, too.

**Comment by hco** [comment]
Managed to add debug logs on my own - I think the following error is the relevant part:

```
time data '8:24' does not match format '%I:%M %p'
```

I cannot believe it, but the response in the harvest API depends on whether you configure your company to use 12 hours or 24 hour format 🙈 I just love their API.

I **could** try to fix this, but I am not really fluent in python, and I think you should give me some pointers on how you'd like it to be fixed. My first try would by to catch that error within `__to_iso_date` and if it fails try it parsing it as 24 hours?
Or we could check the format first and depending on that use a different pattern - easiest way would to look for a space 😂

**Comment by BottlecapDave** [comment]
Thank you for the additional debugging. Yeah their API has a lot to be desired in areas. As I know how to reproduce the issue, I'll have a look over the weekend.

**Comment by BottlecapDave** [comment]
This should be fixed in v1.2.0.

**Comment by hco** [comment]
Wow, that was fast!

just gave it a try, still getting 0 though :(

won‘t be able to debug this until tonight, though

**Comment by BottlecapDave** [comment]
hmmm. I've added some additional logging in the release that will report if it's failing to deserialize the result from harvest. No rush, just when you get a chance :)

**Comment by hco** [comment]
Sorry, took a bit longer. The log message does not tell me an actual error message, but it prints the entries.

What I noticed is that the times are not with a zero padded hours - could that be the issue?

**Comment by BottlecapDave** [comment]
That shouldn't cause an issue. You might need to set the logs to debug mode to get the logging information I'm looking for, if you haven't already. It's in the form 'failed to transform data'.

Also I assume you restarted your HA instance upon update?

**Comment by BottlecapDave** [comment]
I think I found the issue. Might have helped if I actually tested the lack of zero padded hours 🤦

The fix should be available in v1.2.1

**Comment by hco** [comment]
Haha, sorry about that 🙈😂

We‘re getting there: I now have 37,2 hours for todays, which is not correct but at least not 0 ;-)

will provide you with logs later, but wanted to let you know already.

thanks for the great help and fast iteration!

**Comment by hco** [comment]
Aaaah, got it. It's the combined hours of everyone in the team, not just my user, so I guess it's working now and just not doing what I expected. 

Again, thank you for your fast help!

**Comment by BottlecapDave** [comment]
It should only be returning the data associated with your account. Are you an admin or manager by any chance? If so, it might be because I missed a caveat with the API where the data you see depends on your level.

**Comment by hco** [comment]
Yep, that's the case.

The endpoint for fetching time entries supports filtering by user id. There's no "only my time entries" filter, I think. There is a separate endpoint to fetch the current user (https://api.harvestapp.com/v2/users/me)

Just brainstorming possible solutions I see, feel free to just ignore everything I write :-)

- A simple solution would be to have an (optional) field for the user id in the config flow?
- Additionally to the current sensors, create sensors for every available user
- During config flow fetch users, allow selecting one (or even multiple)
  - If multiple users were selected, should those be summed up or treated separately

**Comment by BottlecapDave** [comment]
Yeah I did a little discovery while I was waiting for your response :) I'm aiming for fixing the sensors so they're limited to the users entries, as this was the original intention. 

I've got a solution that I'm currently testing. Just need to test one additional flow. I'll try and get a release out either today or tomorrow.

Thanks for the patience on these cases that were not covered in the accounts I've been trying against.
